### ISSUE TITLE
[HOLD for payment 2024-07-24] Chat - Mentions popup doesn't appear when editing message

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#19643]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login to NewDot
2. Open any chat
3. Send a message
4. Edit message. While editing try to mention any account using @

## Expected Result:
Popup should appear with possible mentions

## Actual Result:
Popup doesn't appear when editing message.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.18.1

**Reproducible in staging?:** Yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://platform.applause.com/services/links/v1/external/e5a721eda7a8d29c80c1f999e9737f19fd8a4ae7358fcf6bb3f3630dd717b4b0

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d781c638f8bf0faa</li>
        <li>Upwork Job ID: 1664725947469553664</li>
        <li>Last Price Increase: 2023-06-02</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @Christinadobrzyn / @Christinadobrzyn</details>

**Comment by dukenv0307** [comment]
Looks like this is a new feature not a bug

**Comment by Christinadobrzyn** [comment]
I agree, looks like a new feature not a bug but I think it's good to push forward. 

Can we combine this issue with this fix?  https://github.com/Expensify/App/issues/19673

**Comment by MariaHCD** [comment]
> Hey @MariaHCD do you think this is good to build? I think this can be external but let me know if you think it should be internal!

I think it makes sense to have this feature but I think the plan for editing messages with mentions is still a WIP. Could you confirm, @puneetlath?

**Comment by puneetlath** [comment]
Ah interesting. It looks like neither emoji auto-suggestion or mentions auto-suggestion currently works while editing a comment. I agree it is something we should add though. 

@stitesExpensify do you know if we have thought about this for emojis at all already?

**Comment by stitesExpensify** [comment]
I think this is pretty much as far as we got with it https://github.com/Expensify/App/pull/14686#issuecomment-1442025502

Basically "Yeah, we should add that". I don't think that the discussion continued though

**Comment by puneetlath** [comment]
I think we should probably wait until we have the inline auto-suggester: https://github.com/Expensify/App/issues/16078

**Comment by MariaHCD** [comment]
Nice, makes sense. Thanks, @puneetlath! Should this be on hold for the inline auto-suggester? Additionally, I'm assuming once that is complete, this issue can be external?

**Comment by Christinadobrzyn** [comment]
Not overdue - Added daily label since payment on PR https://github.com/Expensify/App/issues/16078 is tomorrow

**Comment by Christinadobrzyn** [comment]
waiting

**Comment by Christinadobrzyn** [comment]
based on this [SO](https://stackoverflowteams.com/c/expensify/questions/16630/16633#16633) - I can reach out to our expert contributors to see if anyone can take this

**Comment by puneetlath** [comment]
@Christinadobrzyn this is actually on hold right now.

**Comment by Christinadobrzyn** [comment]
Sorry @puneetlath, I'm confused, I thought this was on hold for https://github.com/Expensify/App/pull/17331 but that PR seems to be in production?

**Comment by puneetlath** [comment]
Ah yes, I see why that is confusing. #16078 will actually take multiple PRs to complete. The first has gone live, but there will probably be a handful more before it is done.

**Comment by MariaHCD** [comment]
Looks like we're still on hold for https://github.com/Expensify/App/issues/16078

**Comment by Christinadobrzyn** [comment]
on hold for https://github.com/Expensify/App/issues/16078

**Comment by Christinadobrzyn** [comment]
monitoring https://github.com/Expensify/App/issues/16078

**Comment by Christinadobrzyn** [comment]
same ^

**Comment by Christinadobrzyn** [comment]
I'm going to be ooo until August 31st so going to unassign and assign a new teammate. 

At this time, we're on hold for https://github.com/Expensify/App/issues/16078

I'll take this when I'm back if it's still open

**Comment by joekaufmanexpensify** [comment]
Sounds good!

**Comment by joekaufmanexpensify** [comment]
Updated labels as this is on hold.

**Comment by joekaufmanexpensify** [comment]
On hold

**Comment by joekaufmanexpensify** [comment]
@Christinadobrzyn I see you're back from OOO, so going to let you take this one back!

**Comment by Christinadobrzyn** [comment]
Still on hold for https://github.com/Expensify/App/issues/16078

Moving to weekly

**Comment by MariaHCD** [comment]
still on hold for https://github.com/Expensify/App/issues/16078

**Comment by Christinadobrzyn** [comment]
still on hold for https://github.com/Expensify/App/issues/16078

**Comment by MariaHCD** [comment]
Looks like we're still on hold for https://github.com/Expensify/App/issues/16078

**Comment by Christinadobrzyn** [comment]
on hold for https://github.com/Expensify/App/issues/16078

**Comment by Christinadobrzyn** [comment]
on hold for https://github.com/Expensify/App/issues/16078

**Comment by Christinadobrzyn** [comment]
on hold for https://github.com/Expensify/App/issues/16078

**Comment by MariaHCD** [comment]
Still on hold

**Comment by Christinadobrzyn** [comment]
on hold - https://github.com/Expensify/App/issues/16078

**Comment by Christinadobrzyn** [comment]
on hold - https://github.com/Expensify/App/issues/16078

**Comment by puneetlath** [comment]
@perunt would you like to take this one? Now that you've added inline auto-suggestion, we can make user mentions, room mentions, and emoji auto-suggestion work in the edit menu flow.

**Comment by perunt** [comment]
Sure, I would like to do it. I'll start on it at the beginning of next week

**Comment by Christinadobrzyn** [comment]
hi @perunt just checking on this - can you provide an update? TY!

**Comment by perunt** [comment]
hey @Christinadobrzyn, the PR is ready and waiting for a review

**Comment by Christinadobrzyn** [comment]
PR https://github.com/Expensify/App/pull/44720 under review!

**Comment by jasperhuangg** [comment]
FYI https://github.com/Expensify/App/issues/45435 was missed during testing, I'll have another contributor handle it

**Comment by Christinadobrzyn** [comment]
PR in Staging - https://github.com/Expensify/App/pull/44720

Getting close!

**Comment by Christinadobrzyn** [comment]
Preparing for payment

Payouts due:

- [ ] Contributor+: $250 @rayane-djouah - are you paid in Upwork?

@rayane-djouah do we need a regression test for this?

**Comment by Christinadobrzyn** [comment]
Ah thanks @rayane-djouah! Let me create an offer in Upwork... here you go - https://www.upwork.com/nx/wm/offer/103238719

**Comment by Christinadobrzyn** [comment]
Payment day is here - payment summary -https://github.com/Expensify/App/issues/19643#issuecomment-2248626177

Regression test - https://github.com/Expensify/Expensify/issues/415160

closing as complete!
